### PR TITLE
JS: add more models for file related sinks and sources

### PR DESCRIPTION
**Pull request by erik-krogh**
Adds new sinks/sources related to NPM libraries that does file operations. 

The sinks are mostly related to `js/path-injection`, but there are also a few other queries that benefit from these added sources/sinks. 

[Here is a LGTM run](https://lgtm.com/query/2206059004819520838/) that shows some (mostly safe) uses of the new sinks.

An [evaluation on relevant queries](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-erik-krogh.northeurope.cloudapp.azure.com_1584241023256) show good performance, and [a single new result](https://lgtm.com/query/2402265770949753113/) that looks like a TP (but is not dangerous). 

I found some other sinks that use some promise flow, I'll try those out after https://github.com/Semmle/ql/pull/3036 has been merged. 

I've chosen not to include the popular [del](https://www.npmjs.com/package/del) library as a sink, as it protects against deleting the current working directory and above.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/Files.qll [hunk 1]
```diff
@@ -106,3 +123,143 @@ private class FastGlobFileNameSource extends FileNameSource {
...
+  RecursiveReadDir() { this = DataFlow::moduleImport("recursive-readdir").getACall() }
+
+  override DataFlow::Node getAPathArgument() { result = getArgument(0) }
+
+  override DataFlow::Node getAFileNameSource() { result = getCallback([1 .. 2]).getParameter(1) }
+}
+
+/**
+ * Models for the NodeJS library `jsonfile`.
+ */
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/frameworks/Files.qll]:
```suggestion
/**
 * Classes and predicates for modelling the `jsonfile` library (https://www.npmjs.com/package/jsonfile).
 */
```

File: javascript/ql/src/semmle/javascript/frameworks/Files.qll [hunk 2]
```diff
@@ -106,3 +123,143 @@ private class FastGlobFileNameSource extends FileNameSource {
...
+private module JSONFile {
+  /**
+   * Reading a JSON file using the `jsonfile` library.
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/frameworks/Files.qll]:
```suggestion
   * A reader for JSON files.
```
(changed style, removed reference to the implied library context)

File: javascript/ql/src/semmle/javascript/frameworks/Files.qll [hunk 3]
```diff
@@ -106,3 +123,143 @@ private class FastGlobFileNameSource extends FileNameSource {
...
+
+    override DataFlow::Node getADataNode() {
+      this.getCalleeName() = "readFile" and result = getCallback([1 .. 2]).getParameter(1)
+      or
+      this.getCalleeName() = "readFileSync" and result = this
+    }
+  }
+
+  /**
+   * Writing a JSON file using the `jsonfile` library.
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/frameworks/Files.qll]:
```suggestion
   * A writer for JSON files.
```

**Comment by erik-krogh** [comment]
> Regarding ProducesFileNames and ProducedFileName, do you have other uses of those classes in the pipeline? Otherwise, it seems like a minor overkill to introduce them for a single use case in RecursiveReadDir.

Yes, I except to use `ProducesFileNames` again later PRs.

**Pull request merged**